### PR TITLE
Add support for Ruby 3

### PR DESCRIPTION
**Pull request by rzane**
### Summary

> Provide a general description of the code changes in your pull
request... were there any bugs you had fixed? If so, mention them. If
these bugs have open GitHub issues, be sure to tag them here as well,
to keep the conversation linked together.

`TestTrack::AssignmentEventJob` expects keyword arguments, but `TestTrack::UnsyncedAssignmentsNotifier` is passing a hash. This PR splats the hash to make this gem compatible with Ruby 3.

### Other Information

> If there's anything else that's important and relevant to your pull
request, mention that information here. This could include
benchmarks, or other information.

> Thanks for contributing to TestTrack!

/domain @Betterment/test_track_core
/platform @Betterment/test_track_core

/no-platform

### CODE REVIEW

File: Appraisals [hunk 1]
```diff
@@ -1,11 +1,3 @@
-appraise 'rails-5-1' do
-  gem 'railties', '~> 5.1.0'
-end
-
-appraise 'rails-5-2' do
-  gem 'railties', '~> 5.2.0'
-end
-
```
Comment (rzane) [inline Appraisals:8]: These versions of Rails don't run on Ruby 3 and they've reached end of life.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -12,11 +12,9 @@ jobs:
     strategy:
       matrix:
         ruby:
-          - '2.6'
-          - '2.7'
+          - '3.0'
+          - '3.1'
```
Comment (rzane) [inline .github/workflows/ci.yml:16]: 2.6 has reached end of life, and so 2.7 will at the end of this month.

**Review by jesseproudman — DISMISSED** [review]
domain lgtm

**Comment by jesseproudman** [comment]
Do we need to bump the version?

https://github.com/Betterment/test_track_rails_client/blob/master/lib/test_track_rails_client/version.rb

**Review by jesseproudman — DISMISSED** [review]
domain lgtm

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -12,11 +12,9 @@ jobs:
     strategy:
       matrix:
         ruby:
-          - '2.6'
           - '2.7'
+          - '3.0'
```
Comment (smudge) [inline .github/workflows/ci.yml:16]: Worth adding 3.1 while we're here?

Comment (rzane) [reply .github/workflows/ci.yml:16]: I was going to, but Rubocop failed with [this issue](https://github.com/rubocop/rubocop/issues/10258). We need to bump Rubocop, which causes new cops to fail. I'd prefer to do that in a separate PR.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -12,11 +12,9 @@ jobs:
     strategy:
       matrix:
         ruby:
-          - '2.6'
```
[the remark this answers is not in the document]
Comment (rzane) [reply .github/workflows/ci.yml:15]: Would that be considered a breaking change? As of right now, we aren't using any functionality that isn't supported in 2.6. Do we need to forbid people using 2.6 from using this new version?

**Review by smudge — APPROVED** [review]
domain LGTM && platform LGTM!

**Review by smudge — APPROVED** [review]
domain LGTM

**Pull request merged**